File: app/src/main/java/com/ntech/fourtop/ui/topusers/TopUsersViewModel.java
package com.ntech.fourtop.ui.topusers;

import androidx.lifecycle.ViewModel;

import com.ntech.fourtop.data.AppRepository;

public class TopUsersViewModel extends ViewModel {

    private AppRepository repository;
    private String userToken = "";

    public TopUsersViewModel() {
        repository = AppRepository.get();
    }

    public void setUserToken(String userToken) {
        this.userToken = userToken;
    }
}

File: app/src/main/java/com/emikhalets/voteapp/view/topusers/TopUsersViewModel.kt
package com.emikhalets.voteapp.view.topusers

import androidx.lifecycle.ViewModel
import com.emikhalets.voteapp.data.AppRepository
import com.emikhalets.voteapp.data.AppRepository.Companion.get

class TopUsersViewModel : ViewModel() {
    private val repository: AppRepository?
    private var userToken: String? = ""
    fun setUserToken(userToken: String?) {
        this.userToken = userToken
    }

    init {
        repository = get()
    }
}